src/compiler/compile/nodes/Element.ts

```diff
@@ -351,6 +368,28 @@ export default class Element extends Node {
 					const match = fuzzymatch(value, aria_roles);
 					component.warn(attribute, compiler_warnings.a11y_unknown_role(value, match));
 				}
+
+				// no-redundant-roles
+				if (this.name === value) {
+					component.warn(attribute, compiler_warnings.a11y_no_redundant_roles(value));
+				} else if (this.name === 'nav' && value === 'navigation') {
+					component.warn(attribute, compiler_warnings.a11y_no_redundant_roles(value));
+				} else if (this.name === 'a' && value === 'link') {
+					component.warn(attribute, compiler_warnings.a11y_no_redundant_roles(value));
+				} else if (this.name === 'fieldset' && value === 'group') {
+					component.warn(attribute, compiler_warnings.a11y_no_redundant_roles(value));
+				} else if (this.name === 'ul' && value === 'list') {
+					component.warn(attribute, compiler_warnings.a11y_no_redundant_roles(value));
+				}
+
+				const is_parent_section_or_article = is_parent(this.parent, ['section', 'article']);
+				if (!is_parent_section_or_article) {
+					if (this.name === 'header' && value === 'banner') {
+						component.warn(attribute, compiler_warnings.a11y_no_redundant_roles(value));
+					} else if (this.name === 'footer' && value === 'contentinfo') {
+						component.warn(attribute, compiler_warnings.a11y_no_redundant_roles(value));
+					}
```

Comment:
Probably same here as well for consistency

Reply:
This is now addressed in bb975e7